include/swift/Runtime/Mutex.h

```diff
@@ -0,0 +1,276 @@
+//===--- Mutex.h - Lockables ------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// TODO
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_BASIC_MUTEX_H
+#define SWIFT_BASIC_MUTEX_H
+
+#include <functional>
+
+namespace swift {
+
+///
+///
+/// TODO
+///
+///
+class Condition {
```

Comment:
Is std::condition_variable_any good enough? I don't know what its performance drawbacks are. That would allow us to use std::lock_guard and std::unique_lock everywhere instead writing our own lock scoping.